### PR TITLE
Fix several trivial problems found by clang 7's scan-build

### PR DESCRIPTION
**Pull request by jepler**
By invoking `scan-build make` in `ports/unix`, a number of diagnostics are found.  This fixes several of the trivial ones.

### CODE REVIEW

File: py/objtype.c [hunk 1]
```diff
@@ -336,8 +336,10 @@ mp_obj_t mp_obj_instance_make_new(const mp_obj_type_t *self, size_t n_args, cons
             mp_obj_t *args2 = m_new(mp_obj_t, 1 + n_args + 2 * n_kw);
             args2[0] = MP_OBJ_FROM_PTR(self);
             memcpy(args2 + 1, args, n_args * sizeof(mp_obj_t));
-            // copy in kwargs
-            memcpy(args2 + 1 + n_args, kw_args->table, 2 * n_kw * sizeof(mp_obj_t));
+            if (n_kw) {
+                // copy in kwargs
+                memcpy(args2 + 1 + n_args, kw_args->table, 2 * n_kw * sizeof(mp_obj_t));
```
Comment (tannewt) [inline py/objtype.c]: What does this do?

Comment (jepler) [reply py/objtype.c]: The diagnostic from clang scan-build was that in the memcpy statement, `kw_args->table` dereferenced null pointer `kw_args`.  If kw_args is null, then n_kw is 0.  By guarding it with the added `if` statement, the problem is resolved and the NULL pointer is not dereferenced at runtime.

Comment (jepler) [reply py/objtype.c]: (and if n_kw is 0 then there's nothing to copy anyway)

Comment (tannewt) [reply py/objtype.c]: Ah, why not check kw_args for NULL then? `memcpy` should handle the n_kw = 0 ok.

**Comment by jepler** [comment]
Sure, I can rewrite it that way

>

File: py/objtype.c (hunk 1 above)
Comment (jepler) [reply py/objtype.c]: done

**Pull request merged**